File: app/src/main/java/be/appfoundry/aipdemo/injection/AppModule.java
package be.appfoundry.aipdemo.injection;

import javax.inject.Singleton;

import be.appfoundry.aipdemo.model.User;
import dagger.Module;
import dagger.Provides;

@Module
public class AppModule {

    @Provides
    @Singleton
    static User provideUser() {
        User user = new User("Siebe", 27);
        return user;
    }

}

File: app/src/main/java/be/appfoundry/aipdemo/injection/AppModule.kt
package be.appfoundry.aipdemo.injection

import be.appfoundry.aipdemo.model.User
import dagger.Module
import dagger.Provides
import javax.inject.Singleton

@Module
class AppModule {

  @Provides
  @Singleton
  fun provideUser(): User = User("Siebe")
}